### PR TITLE
Updates permissions with default one before updating settings

### PR DESCRIPTION
**Pull request by wtekeu**
# Description

Updates permissions with default one before updating settings
This change allow TLs to upload app settings without always getting the prompt notifying of a change when it is only the result of rev change due to the API change listener updating the settings after overwriting the settings with default one.

medic/medic-conf#113

# Code review items

- Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: api/src/services/settings.js [hunk 1]
```diff
@@ -55,6 +57,9 @@ module.exports = {
         if (!doc.settings) {
           doc.settings = {};
         }
+
+        _.defaults(body.permissions, defaults.permissions);
```
Comment (garethbowen) [inline api/src/services/settings.js]: I think this should be done after the replace/extend, ie: just before the put line 68. Then you can do the defaulting straight on to `doc.settings` which most of the time won't do anything.

**Comment by kennsippell** [comment]
@wtekeu All tests are now passing for this change. Would you please review this and make sure everything is as expected? Maybe even do a quick test to ensure it is still fixes the medic-conf issue?  Then let's complete the code review cycle with Gareth.

Please note that the base branch for this is from https://github.com/medic/medic/pull/5945.  Let's wait for that to merge, then rebase this to the new master, and then merge this.

**Comment by wtekeu** [comment]
@kennsippell .. code looks good, I have noticed the improvements that you wrote. Thanks. Also, I have tested and it is running as expected.

Noticed the following in `package.json`:
```
  "devDependencies": {
    "@kennsippell/medic-conf": "^3.0.7",
```

**Comment by newtewt** [comment]
@wtekeu  can you have a look at the conflicts?

**Comment by kennsippell** [comment]
We're going to have to rebase again after https://github.com/medic/medic/issues/5914 merges. Can we AT that change first?

**Pull request merged**